src/stream/src/executor/source/fetch_executor.rs

```diff
@@ -282,12 +312,17 @@ impl<S: StateStore> FsFetchExecutor<S> {
                                     .get(split_id.clone())
```

Comment:
As mentioned in L303, the reader will only read one file (i.e one split). How about replacing the for loop in L310 with an assert to make this assumption explicit?